hazelcast/src/main/java/com/hazelcast/cache/impl/ICacheService.java

```diff
@@ -119,7 +123,8 @@
      */
     CacheOperationProvider getCacheOperationProvider(String cacheNameWithPrefix, InMemoryFormat storageType);
 
-    UUID addInvalidationListener(String cacheNameWithPrefix, CacheEventListener listener, boolean localOnly);
+    CompletableFuture<EventRegistration> addInvalidationListener(String cacheNameWithPrefix, CacheEventListener listener,
```

Comment:
Why not return CompetableFuture<UUId> or even Future<UUID> ? They were returning UUID before, then it should be sufficient tor return ` Future<UUID>` for the async versions.